scripts/install/install_macos.sh

```diff
@@ -448,6 +450,31 @@ set_opamp_secret_key() {
   fi
 }
 
+# If an existing supervisor.yaml is present, ask whether we should do a clean install.
+# Want to avoid inadvertanly overwriting endpoint or secret_key values.
+ask_clean_install() {
+  if [ -f "$SUPERVISOR_YML_PATH" ]; then
+    command printf "${indent}An installation already exists. Would you like to do a clean install? $(prompt n)"
+    read -r clean_install_response
+    clean_install_response=$(echo "$clean_install_response" | tr '[:upper:]' '[:lower:]')
+    case $clean_install_response in
+    y | yes)
+      increase_indent
+      success "Doing clean install!"
+      decrease_indent
+      clean_install="true"
+      ;;
+    *)
+      warn "Doing upgrade instead of clean install"
+      clean_install="false"
+      ;;
+    esac
+  else
+    warn "Previous supervisor config not found, doing clean install"
+    clean_install="true"
+  fi
+}
```

Comment:
Do you think we could find a way to skip this question in non-interactive contexts? (e.g. add a flag to specify without reading from stdin)

Reply:
Yea, I can add a `-c` flag to indicate a clean install w/o doing the interactive

Reply:
Actually looks like we already use that flag, so I'll go with `-i` for install? Open to suggestions on this part

Reply:
Probably fine as long as some flag exists 😆 There are a few "weird" choices for our short flags already.

I'm thinking we might want a way to also force a non-clean install? I'm wondering what you think that'd be useful. I think our automated tests assume you're able to non-interactively do an upgrade without specifying an endpoint.

Reply:
I can add it in pretty quick, so I can just do that, especially if it's something that's needed/wanted for automated testing.